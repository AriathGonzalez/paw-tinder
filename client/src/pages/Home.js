import Nav from "../components/Nav";
import AuthModal from "../components/AuthModal";
import { useState } from "react";
import { useCookies } from "react-cookie";
import "../styles/Homepage.css";

const Home = () => {
  const [showModal, setShowModal] = useState(false);
  const [isSignUp, setIsSignUp] = useState(true);
  const [cookies, , removeCookie] = useCookies(["user"]);
  const authToken = cookies.AuthToken;

  const handleClick = () => {
    if (authToken) {
      removeCookie("UserId", cookies.UserId);
      removeCookie("AuthToken", cookies.AuthToken);
      window.location.reload();
      return;
    }
    setShowModal(true);
    setIsSignUp(true);
  };

  return (
    <div className="overlay">
      <Nav
        authToken={authToken}
        minimal={false}
        showModal={showModal}
        setShowModal={setShowModal}
        setIsSignUp={setIsSignUp}
      />
      <div className="home">
        <h1 className="primary-title">Sniff RightÂ®</h1>
        <button className="primary-button" onClick={handleClick}>
          {authToken ? "Signup" : "Create Account"}
        </button>
        {showModal && (
          <AuthModal
            setShowModal={setShowModal}
            isSignUp={isSignUp}
            setIsSignUp={setIsSignUp}
          />
        )}
      </div>
    </div>
  );
};

export default Home;
