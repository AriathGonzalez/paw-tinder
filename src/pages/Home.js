import Nav from "../components/Nav";
import AuthModal from "../components/AuthModal";
import { useState } from "react";
import "../styles/Homepage.css";

const Home = () => {
  const [showModal, setShowModal] = useState(false);
  const [isSignUp, setIsSignUp] = useState(true);

  const authToken = false;

  const handleClick = () => {
    console.log("home, create account clicked!");
    setShowModal(true);
  };

  return (
    <div className="overlay">
      <Nav
        minimal={false}
        showModal={showModal}
        setShowModal={setShowModal}
        setIsSignUp={setIsSignUp}
      />
      <div className="home">
        <h1 className="primary-title">Sniff RightÂ®</h1>
        <button className="primary-button" onClick={handleClick}>
          {authToken ? "Signup" : "Create Account"}
        </button>
        {showModal && (
          <AuthModal
            setShowModal={setShowModal}
            isSignUp={isSignUp}
            setIsSignUp={setIsSignUp}
          />
        )}
      </div>
    </div>
  );
};

export default Home;
