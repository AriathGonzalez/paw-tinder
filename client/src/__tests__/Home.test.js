import Home from "../pages/Home";
import renderer from "react-test-renderer";

describe("Home Page", () => {
  describe("Snapshot Test", () => {
    it("Should render correctly", () => {
      const tree = renderer.create(<Home />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
