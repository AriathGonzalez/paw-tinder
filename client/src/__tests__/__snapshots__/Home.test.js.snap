// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Snapshot Test Should render correctly 1`] = `
<div
  className="overlay"
>
  <nav>
    <div
      className="logo-container"
    >
      <img
        alt="Logo"
        className="logo"
        src="paw_tinder_logo_black.png"
      />
    </div>
  </nav>
  <div
    className="home"
  >
    <h1
      className="primary-title"
    >
      Sniff RightÂ®
    </h1>
    <button
      className="primary-button"
      onClick={[Function]}
    >
      Create Account
    </button>
  </div>
</div>
`;
